zcash_extras/src/wallet.rs

```diff
@@ -0,0 +1,268 @@
+/// column in order to broadcast the transaction to the network.
+///
+/// Do not call this multiple times in parallel, or you will generate transactions that
+/// double-spend the same notes.
+///
+/// # Transaction privacy
+///
+/// `ovk_policy` specifies the desired policy for which outgoing viewing key should be
+/// able to decrypt the outputs of this transaction. This is primarily relevant to
+/// wallet recovery from backup; in particular, [`OvkPolicy::Discard`] will prevent the
+/// recipient's address, and the contents of `memo`, from ever being recovered from the
+/// block chain. (The total value sent can always be inferred by the sender from the spent
+/// notes and received change.)
+///
+/// Regardless of the specified policy, `create_spend_to_address` saves `to`, `value`, and
+/// `memo` in `db_data`. This can be deleted independently of `ovk_policy`.
+///
+/// For details on what transaction information is visible to the holder of a full or
+/// outgoing viewing key, refer to [ZIP 310].
+///
+/// [ZIP 310]: https://zips.z.cash/zip-0310
+///
+/// # Examples
+///
+/// ```
+/// use tempfile::NamedTempFile;
+/// use zcash_primitives::{
+///     consensus::{self, Network},
+///     constants::testnet::COIN_TYPE,
+///     transaction::components::Amount
+/// };
+/// use zcash_proofs::prover::LocalTxProver;
+/// use zcash_client_backend::{
+///     keys::spending_key,
+///     data_api::wallet::create_spend_to_address,
+///     wallet::{AccountId, OvkPolicy},
+/// };
+/// use zcash_client_sqlite::{
+///     WalletDb,
+///     error::SqliteClientError,
+///     wallet::init::init_wallet_db,
+/// };
+///
+/// # // doctests have a problem with sqlite IO, so we ignore errors
+/// # // generated in this example code as it's not really testing anything
+/// # fn main() {
+/// #   test();
+/// # }
+/// #
+/// # fn test() -> Result<(), SqliteClientError> {
+/// let tx_prover = match LocalTxProver::with_default_location() {
+///     Some(tx_prover) => tx_prover,
+///     None => {
+///         panic!("Cannot locate the Zcash parameters. Please run zcash-fetch-params or fetch-params.sh to download the parameters, and then re-run the tests.");
+///     }
+/// };
+///
+/// let account = AccountId(0);
+/// let extsk = spending_key(&[0; 32][..], COIN_TYPE, account.0);
+/// let to = extsk.default_address().unwrap().1.into();
+///
+/// let data_file = NamedTempFile::new().unwrap();
+/// let db_read = WalletDb::for_path(data_file, Network::TestNetwork).unwrap();
+/// init_wallet_db(&db_read)?;
+/// let mut db = db_read.get_update_ops()?;
+///
+/// create_spend_to_address(
+///     &mut db,
+///     &Network::TestNetwork,
+///     tx_prover,
+///     account,
+///     &extsk,
+///     &to,
+///     Amount::from_u64(1).unwrap(),
+///     None,
+///     OvkPolicy::Sender,
+/// )?;
+///
+/// # Ok(())
+/// # }
+/// ```
+#[allow(clippy::too_many_arguments)]
+pub async fn create_spend_to_address<E, N, P, D, R>(
```

Comment:
As I can see, the original `create_spend_to_address` is not used in Komodefi code. Can this be removed?

Reply:
I'm currently using `create_spend_to_address` is used in Komodefi `WASM` unit test